File: frontend/src/api/news.ts
import { apiClient } from "./client";

export interface NewsArticle {
  title: string;
  url: string;
  source: string;
  published_at: string;
}

export interface NewsSearchRequest {
  query: string;
  max_results?: number;
}

export interface NewsSearchResponse {
  query: string;
  articles: NewsArticle[];
}

export async function searchNews(request: NewsSearchRequest): Promise<NewsSearchResponse> {
  const { data } = await apiClient.post<NewsSearchResponse>("/news/search", request);
  return data;
}

File: backend/app/schemas/news.py
from pydantic import BaseModel, Field


class NewsSearchRequest(BaseModel):
    query: str = Field(min_length=1, max_length=200)
    max_results: int = Field(default=5, ge=1, le=20)


class NewsArticle(BaseModel):
    title: str
    url: str
    source: str
    published_at: str


class NewsSearchResponse(BaseModel):
    query: str
    articles: list[NewsArticle]
